src/main/java/org/jabref/gui/importer/GrobidUseDialogHelper.java

```diff
@@ -7,29 +7,27 @@
 /**
  * Metadata extraction from PDFs and plaintext works very well using Grobid, but we do not want to enable it by default
  * due to data privacy concerns.
- * To make users aware of the feature, we ask each time before querying Grobid, giving the option to opt-out.
+ * To make users aware of the feature, we ask before querying for the first time Grobid, saving the users' preference
  */
-public class GrobidOptInDialogHelper {
+public class GrobidUseDialogHelper {
 
     /**
-     * If Grobid is not enabled but the user has not explicitly opted-out of Grobid, we ask for permission to send data
-     * to Grobid using a dialog and giving an opt-out option.
+     * If the user has not explicitly opted-in/out of Grobid, we ask for permission to send data to Grobid by using
+     * a dialog. The users' preference is saved.
      *
      * @param dialogService the DialogService to use
      * @return if the user enabled Grobid, either in the past or after being asked by the dialog.
      */
     public static boolean showAndWaitIfUserIsUndecided(DialogService dialogService, GrobidPreferences preferences) {
-        if (preferences.isGrobidEnabled()) {
-            return true;
+        if (preferences.isGrobidUseAsked()) {
+            return preferences.isGrobidEnabled();
         }
-        if (preferences.isGrobidOptOut()) {
-            return false;
-        }
-        boolean grobidEnabled = dialogService.showConfirmationDialogWithOptOutAndWait(
+        boolean grobidEnabled = dialogService.showConfirmationDialogAndWait(
```

Comment:
The dialog should not be displayed if preferences are enabled without it. Here's an example:

1. Reset preferences.
2. Restart.
3. Open preferences, then go to Web Search and allow Grobid.
4. Import a file.
5. The dialog will prompt to enable Grobid, even though it is already enabled.

Reply:
Hey @LoayGhreeb, thanks for the suggestion. I have made changes such that if a user has manually enabled Grobid, they will _not_ be prompted in any case (even if they manually disable Grobid in the future) as required.